### DISCUSSION TITLE
Help with Now Playing Search

### DISCUSSION THREAD

**Discussion by fungi1970** [nicotine-plus/nicotine-plus#2380] [Q&A] [answered]
Hello.  New to Nicotine+ and trying to get the Now Playing Search plugin to function.  The plugin is loading according to the log, and Nicotine+ is getting my current play info from ListenBrainz without issue (verified with "test configuration" button in "Now Playing").

Despite everything apparently working, no searches are taking place.  Am just expecting different behavior that what it provides?  

I am running Nicotine+ 3.2.8 on Windows 10 Pro.

Thank you!

**Comment by slook** [comment]
The code in the plugin appears to expect some keyword to be entered (such as `$n` for example) into the search term text entry?

It is reasonable that you would expect it to search when the track changes, but the Now Playing Search plugin doesn't seem to do that.

**Reply by fungi1970** [reply]
Listenbrainz does pass that variable and that can be seen.  Yes, I would expect it to actually search, since that's the name of the plugin.  :)  It is not working, so bug?

**Reply by slook** [reply]
> Listenbrainz does pass that variable and that can be seen

Where do you see it?

**Reply by fungi1970** [reply]
When I click the "Test Configuration".  It is populating fine.

**Reply by slook** [reply]
How about when you enter `$n` into the text entry in Search Files whilst a track is playing?

**Reply by fungi1970** [reply]
Opens a tab that says "Playing now..." under the search, but no results are returned.

**Reply by slook** [reply]
Then it is searching for the string "Playing now...", but I don't understand why there are no results since searching for "Playing now" does return plenty of results.

Either try to change ListenBrainz ouput for the `$n` keyword, or use another keyword like `$t` (for track title)?

**Comment by fungi1970** [comment]
I'll play around with it.  Thanks for pointing me in the right direction.

**Reply by slook** [reply]
Please do let us know what you can come up with. Your feedback would be useful as the Now Playing features are planned for overhaul at some point, but I don't yet known how to approach this since I don't personally use the feature.

**Comment by fungi1970** [comment]
$n does not return results, as it is searching for the entire string, including "Now Playing".  $a $t works and returns results as expected.  It seems to be a code bug that the search function passes the entire string, as opposed to just the criteria that would create a valid search result.

**Reply by slook** [reply]
Maybe they changed it from "Playing now" to "Now playing"

https://github.com/nicotine-plus/nicotine-plus/blob/468288c790a586994619f51e6012f9b3e258ccf7/pynicotine/nowplaying.py#L286

Idk how if this supposed to match the string before replacing the argument or not.

**Comment by fungi1970** [comment]
N+ Python is injecting the "Playing now:" string.  Either it should not add that to the variable, or add a filter to remove "Playing now:" from the variable in the search function, should it be present.

**Comment by fungi1970** [comment]
FYI - I was not looking yesterday when I typed my response, so when I referred to "Now Playing" I actually meant the "Playing now:" string.  The problem is that it is being added to the $n variable, and then  the search function is pulling the entire thing, including "Playing now" and the colon immediately following.  

The quick solution for now is to just paste in "$a $t" into the search, as that returns the expected result.

**Comment by slook** [comment, marked as answer]
Understood, proposed fix in PR https://github.com/nicotine-plus/nicotine-plus/pull/2385

**Comment by slook** [comment]
The fix is now merged into the master branch Nicotine 3.3.0.dev4
https://github.com/nicotine-plus/nicotine-plus/blob/master/doc/TESTING.md#windows
